source/includes/k8s/file-transfer-protocol-k8s.rst

```diff
@@ -165,3 +165,66 @@ If SFTP is enabled, the output resembles the following:
 
    enableSFTP: true
 
+Connect to MinIO Using SFTP with a Certificate Key File
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: RELEASE.2024-05-07T06-41-25Z
+
+MinIO supports user certificate based authentication on SFTP.
+
+This example adds a certificate signature for the MinIO user ``sftp-ca-user1``.
+The signature remains valid for one week after creation.
+
+Before beginning, the following prerequisites must be met:
+
+- Create a trusted user Certificate Authority, such as with ``ssh-keygen -f user_ca``
+- Start or restart the MinIO server to support this CA by including the following flag in the command string:
+
+  .. code-block:: bash
+     :class: copyable 
+
+     --sftp=trusted-user-ca-key=/path/to/.ssh/user_ca.pub
+
+Repeat the following steps for each user who accesses the MinIO Server by SFTP with a user CA key file:
+
+1. Create user public key in client PC (testuser1 in this example) ssh-keygen
+2. Provide copy of /home/testuser1/.ssh/id_rsa.pub to CA server.
+3. Create a signature for the identity ``sftp-ca-user1``.
+   (The name must match the username in MinIO).
+   In this example, the signature is valid for one week.
+   
+   .. code-block:: bash
+      :class: copyable
+
+      ssh-keygen -s /home/miniouser/.ssh/user_ca -I sftp=ca-user1-2024-05-03 -n sftp-ca-user1 -V +1w id_rsa.pub
+
+4. Copy ``id_rsa-cert.pub`` to ``/home/sftp-ca-user1/.ssh/id_rsa-cert.pub`` on the client PC.
```

Comment:
I think users are going to get confused as hell in general with all the public keys and signatures that have to be generated.

Also step 2 - not sure what it means to provide it to 'CA server'.

From the setting:

" The user can then use the passed SSH Public Key to log in to SFTP servers. "

That implies that the AD/LDAP server is going to return some sort of public key as part of the user auth here too. How does that play in?

My guess - this requires AD/LDAP server to already have the user's public key saved. MinIO compares the local public key to the remote one, and as long as we trust the CA used to mint that key, we can allow SFTP to happen.

I'd check w/ engineering -the steps above are good for testing/evaluation but I think we'd have to revisit them since there is likely an AD/LDAP piece missing from the equation.

Reply:
Reworked this section entirely. The key file authentication has nothing to do with LDAP. It's an alternative way of authenticating.